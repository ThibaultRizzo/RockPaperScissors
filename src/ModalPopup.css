.modal-background.full-page-background {
  display: flex;
  flex-flow: row nowrap;
  justify-content: center;
  align-items: center;

  background-color: rgba(0, 0, 0, 0.4);
}

.modal-popup {
  width: 25%;
  height: 15%;
  background-color: white;
  font-weight: bold;
  font-size: 30px;
  border: none;
  box-shadow: white 2px 0px 2px 5px;
  border-radius: 5px;

  display: flex;
  flex-flow: row nowrap;
  align-items: center;
  justify-content: center;
}

.win {
  color: green;
}

.lose {
  color: red;
}

.draw {
  color: darkblue;
}
