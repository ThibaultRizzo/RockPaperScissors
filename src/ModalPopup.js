import React, { Component } from 'react';
import './ModalPopup.css';
import { Result } from './Rules.js';

function ResultComponent(props) {
  let result;
  if (!props.result) {
    result = false;
  } else if (Result.WIN === props.result) {
    result = <p className="win">You won !</p>;
  } else if (Result.LOSE === props.result) {
    result = <p className="lose">You lost !</p>;
  } else {
    result = <p className="draw">It's a draw !</p>;
  }
  return result;
}

class ModalPopup extends Component {
  constructor(props) {
    super(props);
    this.closePopup = this.closePopup.bind(this);
  }

  /**
   * Closes popup after one second
   */
  closePopup() {
    setTimeout(this.props.closePopup, 1000);
  }

  render() {
    const result = this.props.result;
    return (
      <div
        className="modal-background smooth-appear full-page-background"
        onClick={this.closePopup()}
      >
        <div className="modal-popup">
          <ResultComponent result={result} />
        </div>
      </div>
    );
  }
}

export default ModalPopup;
