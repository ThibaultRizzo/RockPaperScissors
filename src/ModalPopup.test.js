
import React from 'react';
import ReactDOM from 'react-dom';
import ModalPopup from './ModalPopup';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<ModalPopup />, div);
  ReactDOM.unmountComponentAtNode(div);
});
