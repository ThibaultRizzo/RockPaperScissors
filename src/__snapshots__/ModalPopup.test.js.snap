// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="modal-background smooth-appear full-page-background"
  onClick={[Function]}
>
  <div
    className="modal-popup"
  >
    <p
      className="lose"
    >
      You lost !
    </p>
  </div>
</div>
`;
